### PR TITLE
ome-files 0.5.0

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Update versions and add post.

### CODE REVIEW

**Comment by hflynn** [comment]
Building on snoopy now

**Comment by hflynn** [comment]
Staged at https://snoopycrimecop.github.io/www.openmicroscopy.org/2017/12/01/ome-files-0-5-0.html

**Comment by hflynn** [comment]
I'll open a separate PR to fix the docs links since it doesn't touch the same pages

File: _posts/2017-12-01-ome-files-0-5-0.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- Only run thread tests with CMake ≥3.8
+
+*OME Common*
+
+- Conditionally add `Boost::log` and `Boost::log_setup` to exported configuration
+
+*OME Model*
+
+- Add missing explicit `<sstream>` include
+- Remove REQUIRED from exported cmake configuration
```
Comment (hflynn) [inline _posts/2017-12-01-ome-files-0-5-0.md]: Added OMERO pyramid format documentation too

File: _posts/2017-12-01-ome-files-0-5-0.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  - png 1.6.34: Updated package
+  - pybind11 2.2.1: New package
+  - tiff 4.0.9: Updated package
+  - yaml-cpp 0.5.3: New package
+  - all python-* packages are now installed with pip
+
+**Errata**
+
+See the
+[errata](https://docs.openmicroscopy.org/ome-cmake-superbuild/0.5.0/errata.html)
```
Comment (hflynn) [inline _posts/2017-12-01-ome-files-0-5-0.md]: This link should be http://docs.openmicroscopy.org/ome-files-cpp/0.5.0/ome-cmake-superbuild/manual/html/errata.html

File: _posts/2017-12-01-ome-files-0-5-0.md (hunk 1 above)
Comment (rleigh-codelibre) [reply _posts/2017-12-01-ome-files-0-5-0.md]: I've limited the items here to C++-related changes.

File: _posts/2017-12-01-ome-files-0-5-0.md (hunk 2 above)
Comment (rleigh-codelibre) [reply _posts/2017-12-01-ome-files-0-5-0.md]: Fixed.

**Comment by hflynn** [comment]
You've got Dec 1st as the post date too, it's the 4th today

**Comment by rleigh-codelibre** [comment]
The post date was intentional; this is the date the releases were tagged and the release builds made.

**Comment by hflynn** [comment]
In previous cases, we have used the announcement date for these even if the release was tagged previously

File: _posts/2017-12-01-ome-files-0-5-0.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The main change in this release is the re-addition of compatibility
+code to permit building on CentOS6 and RHEL6 with EPEL.  Other work
+includes the switch to C++11 threads and mutexes from the Boost
+equivalents, and preliminary CMake support for building with C++17
+compilers.
+
+**Downloads**
+
+The software release is available at [archived downloads](https://downloads.openmicroscopy.org/ome-files-cpp/0.5.0/). This
```
Comment (sbesson) [inline _posts/2017-12-01-ome-files-0-5-0.md]: We do not longer have a top-level `index.html` there so the links section probably does not hold. Maybe reduce this section to a single sentence pointing to the raw downloads?

File: _posts/2017-12-01-ome-files-0-5-0.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+OME-TIFF file format for the storage and interchange of biological
+imaging data and metadata.  It provides support for:
+
+- the OME Data Model (model objects, metadata store, XSL transforms
+  and XML validation)
+- OME-TIFF reading
+- OME-TIFF writing
+
+The main change in this release is the re-addition of compatibility
+code to permit building on CentOS6 and RHEL6 with EPEL.  Other work
```
Comment (sbesson) [inline _posts/2017-12-01-ome-files-0-5-0.md]: Do we want to mention you can build on CentOS 7 with the package dependencies? CentOS 6 still requires the superbuild right?

Comment (rleigh-codelibre) [reply _posts/2017-12-01-ome-files-0-5-0.md]: I'm not 100% sure.  We haven't tested CentOS 6 with EPEL, but it might work.  I can't remember if there were any blockers on CentOS 6.  I can certainly reword it to remove CentOS 6 here until we have confirmed this.

Comment (sbesson) [reply _posts/2017-12-01-ome-files-0-5-0.md]: :+1: for putting the emphasis on CentOS 7 as we have definitely tested it.

**Comment by sbesson** [comment]
Also looking at the implicit mention of the `ome-files-performance` via the tiling benchmark documentation, should we mention the work/state of `ome-files-py`?

Despite it not being a first-class component of OME Files for this release, we could consider a small sentence mentioning its existence. /cc @joshmoore @simleo

**Comment by rleigh-codelibre** [comment]
@sbesson ome-files-py was mentioned in the 0.4.0 release notes.  There are no ome-files-py changes to reference here.  Well, there are the ome-tiffwriter additions, but there's no release for that to reference is there?

**Comment by sbesson** [comment]
Aha https://www.openmicroscopy.org/2017/07/04/ome-files-cpp-0-4-0.html that's correct thanks. I thought you guys released `ome-files-py` 0.2.0 a few days ago with the tiff writer additions? Adding it to the package changes like we did previously would work.

**Comment by hflynn** [comment]
https://snoopycrimecop.github.io/www.openmicroscopy.org/2017/12/04/ome-files-0-5-0.html updated, looks like we are good to go now if you're satisfied too @sbesson ?

I notice we don't have a link to the release announcement on the downloads page but I can add that on a following PR once this is merged.

File: _posts/2017-12-04-ome-files-0-5-0.md [hunk 5]
```diff
@@ -72,7 +72,7 @@ including Windows, Linux and MacOS X.
   - ome-model 5.6.0: Updated package
   - ome-files 0.5.0: Updated package
   - ome-files-performance: New experimental package (git only)
-  - ome-files-py: Experimental package now include OME-TIFF writer
+  - ome-files-py: Experimental package now includes OME-TIFF writer
     support.
```
Comment (hflynn) [inline _posts/2017-12-04-ome-files-0-5-0.md]: There's an odd full stop here too

**Comment by hflynn** [comment]
👍 Just waiting for everything to be green

File: _posts/2017-12-04-ome-files-0-5-0.md [hunk 6]
```diff
@@ -61,6 +61,10 @@ including Windows, Linux and MacOS X.
 - Remove REQUIRED from exported cmake configuration
 - Add Dockerfile for building with Travis
 
+*OME File Python (Experimental)*
```
Comment (hflynn) [inline _posts/2017-12-04-ome-files-0-5-0.md]: File**s**

Comment (rleigh-codelibre) [inline _posts/2017-12-04-ome-files-0-5-0.md]: Files.

**Comment by hflynn** [comment]
Good to merge now?
